client/scripts/models/DashboardActivityNotification.ts

```diff
@@ -0,0 +1,73 @@
+import { IChainEventData, SupportedNetwork } from '@commonwealth/chain-events';
+import { NodeInfo, Notification, ChainEvent, NotificationSubscription, ChainEventType } from 'models';
+import moment from 'moment';
+
+class DashboardActivityNotification {
+    // Thread and Comment Notifications
+    public readonly categoryId?: string;
+    public readonly viewCount?: number;
+    public readonly likeCount?: number;
+    public readonly commentCount?: number;
+    public readonly createdAt: moment.Moment;
+    public readonly notification_data?: string;
+    public readonly thread_id?: string;
+    private _isRead?: boolean;
+
+    public get isRead(): boolean {
+        return this._isRead;
+    }
+
+    // Chain Event Notifications
+    public readonly typeId?: ChainEventType;
+    public readonly blockNumber?: number; 
+    public readonly eventData?: IChainEventData;
+    public readonly chainEventId?: number;
+    public readonly updatedAt?: moment.Moment;
+    public readonly eventNetwork?: SupportedNetwork;
+    public readonly chain?: string;
+    public readonly icon_url?: string;
+    
+
+    constructor(createdAt, thread_id?, category_id?, notification_data?, viewCount?, likeCount?, commentCount?, isRead?, type_id?, block_number?, event_data?, id?, updated_at?, event_network?, chain?, icon_url?) {
```

Comment:
Can we (1) make all the frontend-scoped code camelCase, (2) use an object as argument for the constructor with types? Makes things much safer since it order-proofs the argument passing. That way, if args/params get out of order, it won't corrupt the data. See OffchainThreads model for an example.

Reply:
Link to relevant docs: https://www.notion.so/hicommonwealth/Dev-Docs-Developer-Site-User-Guide-139ec37b34e048d1a712567ec9d6f15f#b001f8a2775d4f0f94526c2d05b6b295

Reply:
On it!